src/Components/Patient/SampleViewAdmin.tsx

```diff
@@ -121,14 +82,22 @@ export default function SampleViewAdmin() {
       sampleData.date_of_result = new Date().toISOString();
     }
     const statusName = SAMPLE_TEST_STATUS.find((i) => i.id === status)?.desc;
-    const res = await dispatch(patchSample(sampleData, { id: sample.id }));
-    if (res && (res.status === 201 || res.status === 200)) {
-      Notification.Success({
-        msg: `Success - ${statusName}`,
-      });
-      callFetchData(!fetchFlag);
-    }
-    dismissUpdateStatus();
+
+    await request(routes.patchSample, {
+      pathParams: {
+        id: sample.id || 0,
+      },
+      body: sampleData,
+      onResponse: ({ res }) => {
+        if (res?.ok) {
+          Notification.Success({
+            msg: `Success - ${statusName}`,
+          });
+          callFetchData(!fetchFlag);
```

Comment:
Remove this state and call the refetch instead.

```suggestion
```

Reply:
my bad the variable 'fetchFlag' is not used anywhere in the code and can therefore be removed.